pkg/apis/core/validation/cloudprofile.go

```diff
@@ -218,6 +234,27 @@ func validateMachineImages(machineImages []core.MachineImage, fldPath *field.Pat
 		allErrs = append(allErrs, field.Required(fldPath, "must provide at least one machine image"))
 	}
 
+	allErrs = append(allErrs, validateSpecStaticMachineImages(machineImages, fldPath)...)
+
+	for i, image := range machineImages {
+		idxPath := fldPath.Index(i)
+		for index, machineVersion := range image.Versions {
+			versionsPath := idxPath.Child("versions").Index(index)
+			allErrs = append(allErrs, validateContainerRuntimesInterfaces(machineVersion.CRI, versionsPath.Child("cri"))...)
+			allErrs = append(allErrs, validateExpirableVersionInContext(machineVersion.ExpirableVersion, helper.ToExpirableVersions(image.Versions), versionsPath)...)
+		}
+	}
+
+	return allErrs
+}
+
+func validateSpecStaticMachineImages(machineImages []core.MachineImage, fldPath *field.Path) field.ErrorList {
```

Comment:
Same here about the "static" term and moving this to `utils.go`.

Reply:
moves as `ValidateMachineImages`